src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -378,7 +378,7 @@ public void WriteBindings(IResource resource)
 
                     // Check whether the project view this endpoint as Default (for its scheme).
                     // If so, we don't specify the target port, as it will get one from the deployment tool.
-                    (ProjectResource project, string uriScheme, null, _) when !schemesEncountered.Contains(uriScheme) => null,
+                    (ProjectResource project, string uriScheme, null, _) when uriScheme is "http" or "https" && !schemesEncountered.Contains(uriScheme) => null,
```

Comment:
Maybe a bit better to do the http/https test at the time we add to the set? The set is really `schemesEncounteredAmongHttpAndHttps`